### ISSUE TITLE
XHDL/XSIM Linter files in my local directory

### ISSUE THREAD

**Issue by pintert3** [TerosTechnology/vscode-terosHDL#438]
I'm getting `xsim.dir` and `xhdl.pb` files in my local directory when I create a VHDL file. The issue seems similar to issue #98 . It's pretty annoying, cause they are created in every directory I create the files in.

**To Reproduce**
Set Vivado to be the linter and create a new project within which you create some VHDL files

 - OS: Windows 11
 - VSCode version 1.80.2

![image](https://github.com/TerosTechnology/vscode-terosHDL/assets/26062452/4ea00b7d-a873-4c66-884a-cc2bf1e45eda)

**Comment by qarlosalberto** [comment]
Fixed in the new version (5.0.8). Feel free to re-open the issue if the problem continues.

**Comment by Mystzl** [comment]
Still get this problem.
OS: windows 10
VsCode ver: 1.81.0
terosHDL ver: 5.0.9

**Comment by qarlosalberto** [comment]
Can you double check it? Close VSCode, remove the Vivado files and re-open VSCode.

**Comment by Mystzl** [comment]
Yes, after I close VSCode, remove the Vivado files, re-open VSCode, re-open the existing VHDL files, the "xsim.dir" is created again.

**Comment by pintert3** [comment]
Ya, I can also confirm it's still there.
![image](https://github.com/TerosTechnology/vscode-terosHDL/assets/26062452/3d0539f6-4839-43cd-aa24-c31800254afa)

**Comment by qarlosalberto** [comment]
Can you try with this version? https://github.com/TerosTechnology/vscode-terosHDL/releases/tag/latest

**Comment by pintert3** [comment]
Still does it. 
![image](https://github.com/TerosTechnology/vscode-terosHDL/assets/26062452/7ba7716b-c526-4760-a898-765b89a6ba6e)

**Comment by qarlosalberto** [comment]
What is your OS? Only xsim.dir or also with the other files?

**Comment by pintert3** [comment]
Windows 11. And yes only `xsim.dir` this time

**Comment by qarlosalberto** [comment]
Try again: https://github.com/TerosTechnology/vscode-terosHDL/releases/tag/latest

**Comment by pintert3** [comment]
Sorry, didn't reply in time, I must have forgotten to respond. It still happens. Only `xsim.dir` in every folder I open a vhdl file in.

**Comment by qarlosalberto** [comment]
Try this version: https://github.com/TerosTechnology/vscode-terosHDL/releases/tag/latest

I think that it's a bug in the windows delete command for directories

**Comment by pintert3** [comment]
The `xsim.dir` still appears.

**Comment by qarlosalberto** [comment]
I don't know what happen... If you type in the cmd: `del -Recurse xsim.dir`. Can you remove that folder? Because I'm running that command...

**Comment by pintert3** [comment]
Right, so I tried that command: `del -Recurse xsim.dir` and it works....only in powershell. When I run it in the usual Windows CMD prompt, it doesn't work, it throws an error: 

```

C:\Users\Asus\Desktop\tests\teros\hello>del -Recurse xsim.dir
Could Not Find C:\Users\Asus\Desktop\tests\teros\hello\-Recurse
C:\Users\Asus\Desktop\tests\teros\hello\xsim.dir\*, Are you sure (Y/N)? y

```
And in the end, does nothing.

That gives me the idea that the command could be running in the command prompt shell instead of Powershell.
To remove it totally in the command prompt I'd use:
```
rmdir /s /q xsim.dir
```
And to do the same but in powershell, I'd use your command:
```
del -Recurse xsim.dir
```

**Comment by qarlosalberto** [comment]
Added `rmdir`: https://github.com/TerosTechnology/vscode-terosHDL/releases/tag/latest

**Comment by pintert3** [comment]
So it doesn't work. but I've taken a look at the commit, and seen the command being run, in particular this part
```
... del ${path_3} && del -Recurse ${path_4} && Remove-Item -Recurse -Path ${path_4} && rmdir /s /q ${path_4}
```

**TL;DR** : Replace the above snippet with 
```
... del ${path_3} && rmdir /s /q ${path_4}
```

Here's an explanation if you might need it:
So for starters, I made a small test directory structure to test the command results. The directory structure is like so:
```
    Directory: C:\Users\Asus\Desktop\mine\teros\test_deletion


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         8/21/2023   5:57 PM                xhdl.dir
-a----         8/21/2023   5:56 PM              0 xvlog.pb
```
So of course I substitute the `${path_3}` with `xvlog.pb`, which is a file, and `${path_4}` with `xhdl.dir` which is a directory of course.
Running the command with powershell brought me this:
```ps1
PS C:\Users\Asus\Desktop\mine\teros\test_deletion> del xvlog.pb && del -Recurse xhdl.dir && Remove-Item -Recurse -Path xhdl.dir && rmdir /s /q xhdl.dir
At line:1 char:14
+ del xvlog.pb && del -Recurse xhdl.dir && Remove-Item -Recurse -Path x ...
+              ~~
The token '&&' is not a valid statement separator in this version.
At line:1 char:39
+ del xvlog.pb && del -Recurse xhdl.dir && Remove-Item -Recurse -Path x ...
+                                       ~~
The token '&&' is not a valid statement separator in this version.
At line:1 char:78
+ ... l -Recurse xhdl.dir && Remove-Item -Recurse -Path xhdl.dir && rmdir / ...
+                                                                ~~
The token '&&' is not a valid statement separator in this version.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvalidEndOfLine
```
And didn't delete anything at all. Which to me implies that the command is running in a windows cmd shell. Now on running the command in a cmd shell: 
```bat
C:\Users\Asus\Desktop\mine\teros\test_deletion>del xvlog.pb && del -Recurse xhdl.dir && Remove-Item -Recurse -Path xhdl.dir && rmdir /s /q xhdl.dir
Could Not Find C:\Users\Asus\Desktop\mine\teros\test_deletion\-Recurse
C:\Users\Asus\Desktop\mine\teros\test_deletion\xhdl.dir\*, Are you sure (Y/N)? ls
C:\Users\Asus\Desktop\mine\teros\test_deletion\xhdl.dir\*, Are you sure (Y/N)? y
'Remove-Item' is not recognized as an internal or external command,
operable program or batch file.
```
Now, according to this output, the first command runs just fine, but the second one brings up the issue in my previous comment, where no files are eventually deleted because `xhdl.dir` is a directory. The third fails cause CMD doesn't know of a `Remove-Item` command, and because of the nature of `&&` (versus `||`), the last command doesn't run at all, which explains why only `xhdl.dir` remains.

In conclusion, the solution is to remove all commands attempting to remove `${path_4}` and replace them with `rmdir /s /q ${path_4}` .

**Comment by qarlosalberto** [comment]
Can you try to replace && by &? I need to do it compatible with both, CMD
and powershell

**Comment by pintert3** [comment]
Using `&` the way you're suggesting doesn't work too. It's reserved for future use.
Alternatively, a work around I have in mind is to run the powershell command through CMD like so:
```powershell
cmd /c "del ${path_1} && del ${path_2} ..."
```

It should also work for CMD I believe.

**Comment by qarlosalberto** [comment]
I have moved it to a native implementation, plase check again: 

https://github.com/TerosTechnology/vscode-terosHDL/releases/tag/latest

**Comment by pintert3** [comment]
It works perfectly now. Thanks!

**Comment by qarlosalberto** [comment]
Finally! 😅
